src/ln/channel.rs

```diff
@@ -2943,17 +2943,19 @@ impl Channel {
 			}
```

Comment:
holding_cells are being drained here. Do we forgo ClaimHTLC(payment_preimage) here?

Reply:
Yeah, in fact modifications of channel force_shutdown were due to first design with unsolved htlcs. I delete this stuff thank to better get around with locks in block_connected. So even if I reverted it to original state as of aed435a, I think that the case described by TODO is gone thanks to registering previous inbound htlcs in claimable_htlcs at process_pending_forward ?